File: QuantumsoftHrms/src/app/services/compliance-alert.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({ providedIn: 'root' })
export class ComplianceAlertService {

  private baseUrl = '/api/compliance/alerts';

  constructor(private http: HttpClient) {}

  getAllAlerts(): Observable<any[]> {
    return this.http.get<any[]>(this.baseUrl);
  }

  getUpcomingAlerts(days: number): Observable<any[]> {
    const params = new HttpParams().set('days', days);
    return this.http.get<any[]>(`${this.baseUrl}/upcoming`, { params });
  }

  getAlertHistory(filters: any): Observable<any[]> {
    let params = new HttpParams();
    if (filters.status) params = params.set('status', filters.status);
    if (filters.channel) params = params.set('channel', filters.channel);
    if (filters.fromDate) params = params.set('fromDate', filters.fromDate);
    if (filters.toDate) params = params.set('toDate', filters.toDate);

    return this.http.get<any[]>(`${this.baseUrl}/history`, { params });
  }

  createAlert(payload: any): Observable<any> {
    return this.http.post(this.baseUrl, payload);
  }

  updateAlert(alertId: number, payload: any): Observable<any> {
    return this.http.put(`${this.baseUrl}/${alertId}`, payload);
  }

  deleteAlert(alertId: number): Observable<any> {
    return this.http.delete(`${this.baseUrl}/${alertId}`, { responseType: 'text' });
  }
}

File: backend/src/main/java/com/example/User/service/ComplianceAlertService.java
package com.example.User.service;

import com.example.User.dto.ComplianceAlertRequestDto;
import com.example.User.entity.ComplianceAlert;

import java.time.LocalDate;
import java.util.List;

public interface ComplianceAlertService {
    ComplianceAlert createAlert(ComplianceAlertRequestDto request);

    ComplianceAlert getAlertById(Long alertId);

    List<ComplianceAlert> getAllAlerts();

    List<ComplianceAlert> getUpcomingAlerts(int days);

    List<ComplianceAlert> getAlertHistory(String status, String channel, LocalDate fromDate, LocalDate toDate);

    boolean deleteAlert(Long alertId);

    ComplianceAlert updateAlert(Long alertId, ComplianceAlertRequestDto request);
}
